src/schema.rs

```diff
@@ -433,8 +498,21 @@ impl Schema {
             "float" => Ok(Schema::Float),
             "bytes" => Ok(Schema::Bytes),
             "string" => Ok(Schema::String),
-            other => Err(Error::ParsePrimitive(other.into())),
+            _ => self.fetch_schema(name),
+        }
+    }
+
+    fn fetch_schema(&mut self, name: &str) -> AvroResult<Schema> {
```

Comment:
Would it make sense to add a little bit of documentation to quickly explain what this is doing?
The code speaks for itself, but it may help to reduce cognitive load when reading it 😄

The name of this method makes it seems as if it is stateless (`fetch_`), but it can actually remove things from `input_schemas`.